test/e2e/apps/replica_set.go

```diff
@@ -340,3 +349,72 @@ func testRSAdoptMatchingAndReleaseNotMatching(f *framework.Framework) {
+
+	ns := f.Namespace.Name
+	c := f.ClientSet
+
+	// Create webserver pods.
+	rsPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  WebserverImageName,
+	}
+
+	rsName := "test-rs"
+	replicas := int32(1)
+	rsRevision := "3546343826724305832"
+	annotations := make(map[string]string)
+	annotations[deploymentutil.RevisionAnnotation] = rsRevision
+	rs := newRS(rsName, replicas, rsPodLabels, WebserverImageName, WebserverImage, nil)
+	rs.Annotations = annotations
+	framework.Logf("Creating replica set %q (going to be adopted)", rs.Name)
+	_, err := c.AppsV1().ReplicaSets(ns).Create(context.TODO(), rs, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+	// Verify that the required pods have come up.
+	err = e2epod.VerifyPodsRunning(c, ns, "sample-pod", false, replicas)
+	framework.ExpectNoError(err, "error in waiting for pods to come up: %s", err)
+
+	ginkgo.By("getting scale subresource")
+	scale, err := c.AppsV1().ReplicaSets(ns).GetScale(context.TODO(), rsName, metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to get scale subresource: %v", err)
+	}
+	framework.ExpectEqual(scale.Spec.Replicas, int32(1), "")
+	framework.ExpectEqual(scale.Status.Replicas, int32(1), "")
+
+	ginkgo.By("updating a scale subresource")
+	scale.ResourceVersion = "" // indicate the scale update should be unconditional
+	scale.Spec.Replicas = 2
+	scaleResult, err := c.AppsV1().ReplicaSets(ns).UpdateScale(context.TODO(), rsName, scale, metav1.UpdateOptions{})
+	if err != nil {
+		framework.Failf("Failed to put scale subresource: %v", err)
+	}
+	framework.ExpectEqual(scaleResult.Spec.Replicas, int32(2), "")
+
+	ginkgo.By("verifying the replicaset Spec.Replicas was modified")
+	rs, err = c.AppsV1().ReplicaSets(ns).Get(context.TODO(), rsName, metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to get statefulset resource: %v", err)
+	}
+	framework.ExpectEqual(*(rs.Spec.Replicas), int32(2), "")
+
+	ginkgo.By("Patch a scale subresource")
+	scale.ResourceVersion = "" // indicate the scale update should be unconditional
+	scale.Spec.Replicas = 4    // should be 2 after "UpdateScale" operation, now Patch to 4
+	rsScalePatchPayload, err := json.Marshal(autoscalingv1.Scale{
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: scale.Spec.Replicas,
+		},
+	})
+	framework.ExpectNoError(err, "Could not Marshal JSON for patch payload")
+
+	_, err = c.AppsV1().ReplicaSets(ns).Patch(context.TODO(), rsName, types.StrategicMergePatchType, []byte(rsScalePatchPayload), metav1.PatchOptions{}, "scale")
+	framework.ExpectNoError(err, "Failed to patch replicaset: %v", err)
+
+	ginkgo.By("verifying the replicaset Spec.Replicas was modified")
+	rs, err = c.AppsV1().ReplicaSets(ns).Get(context.TODO(), rsName, metav1.GetOptions{})
+	framework.ExpectNoError(err, "Failed to get replicaset resource: %v", err)
+	framework.ExpectEqual(*(rs.Spec.Replicas), int32(4), "replicaset should have 4 replicas")
+	e2ereplicaset.WaitForReplicaSetTargetAvailableReplicas(c, rs, 4)
```

Comment:
Similarly to the other PR, we don't care about the result. This test is only ensuring the patch operation worked as expected, there's a separate test exercising scaling capabilities.

Reply:
Updated as requested and rebased
Thank you for the review

Reply:
@soltysh are you saying that we should keep the Get and verify `rs.Spec.Replicas` is 4 ... But drop the `WaitForReplicaSetTargetAvailableReplicas`? looks like latest commit does not have the former.

Reply:
@dims that is correct, the test itself is ensuring that update/patch operation on scale resource succeeded. We have a separate test ensuring that scale operation works as a whole.